src/EFCore/Query/Internal/NavigationExpandingExpressionVisitor.ExpressionVisitors.cs

```
@@ -143,6 +143,9 @@ protected override Expression VisitMethodCall(MethodCallExpression methodCallExp
                     return ownedExpansion;
                 }
 
+                // create a dummy query root to make sure that we can actually expand this navigation (later)
+                var _ = _extensibilityHelper.CreateQueryRoot(targetType, entityReference.QueryRootExpression);
```

feels hacky - we could introduce a dedicated checking method on the extensibility helper (so we can avoid the root creation itself), but that checking method would need to throw itself, rather than just returning false if we can't expand temporal table due to range/all operation @smitpatel

---

Adding bool method is fine.